### DISCUSSION TITLE
browser back button does not re-render component after router.push

### DISCUSSION THREAD

**Discussion by sukhsfu** [vercel/next.js#35238] [Help]
Scenario :
I am on Page A which have a X component that has state `sunny ` = false initially. after I did some operation in component; state is updated to `sunny` = true. now I route.push(Page B). when I navigate back to Page A using browser back button I see component X has stale state 'sunny' = true. I want it to reset to `sunny` = false.

Is it bug in nextjs route? If not how can I force particular component to re-render without using stale states

**Comment by icyJoseph** [comment]
Hi,

I've the following:

```js
// pages/index.js
import { useReducer } from "react";
import { useRouter } from "next/router";

export default function Home() {
  const [value, toggle] = useReducer((x) => !x, false);
  const router = useRouter();

  return (
    <div>
      <button onClick={toggle}>toggle</button>
      <span>{value ? "ON" : "OFF"}</span>
      <nav>
        {/* Normally I'd use Link with a tag instead, but you specifically said router.push */}
        <button onClick={() => router.push("/post")}>Post</button>
      </nav>
    </div>
  );
}
```

And

```js
// pages/post
export default function Post() {
  return <div>POST</div>;
}
```

And the `value` behaves as expected. I toggle it ON, navigate to Post, click on the browser back button, and value is back at `OFF`.

I guess you are either doing some kind of mutation? or perhaps shared layouts, which persist across pages.

Could you please share more code?
